Assets/Scripts/Enemy/Minion/MinionController.cs

```diff
@@ -26,39 +28,50 @@ public class MinionController : MonoBehaviour
     [SerializeField] private Transform launchPoint;
     [SerializeField] private Quaternion launchAngle;
 
-    public void Start(){
+    public float currentHealth;
+
+    public void Start()
+    {
         rb = GetComponent<Rigidbody2D>();
-        if(projectile == null || launchPoint == null){
+        if (projectile == null || launchPoint == null)
+        {
             Debug.Log("Minion Controller: projectile spawn parameters null -- component disabled...");
             this.gameObject.SetActive(false);
             return;
         }
-        if(this.transform.position != anchor.transform.position){
+        if (this.transform.position != anchor.transform.position)
+        {
             StartCoroutine(moveToAnchor());
         }
-
+        currentHealth = maxhealth;
         burstTimer = burstDensity;
         return;
     }
 
-    public void Update(){
-        if(attackTimer >= attackPeriod){
+    public void Update()
+    {
+        if (attackTimer >= attackPeriod)
```

Comment:
Could you please also add a quick check so the minion only starts attacking after it reaches its anchor point?